include/magic_enum.hpp

```diff
@@ -363,23 +396,24 @@ inline constexpr auto type_name_v = n<E>();
 template <typename E, E V>
 constexpr auto n() noexcept {
   static_assert(is_enum_v<E>, "magic_enum::detail::n requires enum type.");
-  [[maybe_unused]] constexpr auto custom_name = customize::enum_name<E>(V);
+  using std::holds_alternative;
+  using std::get;
 
-  if constexpr (custom_name.empty()) {
-    if constexpr (supported<E>::value) {
+  [[maybe_unused]] constexpr auto custom = customize::enum_name<E>(V);
```

Comment:
probably a static_assert check is necessary here to make sure to exactly return with the expected variant type.
`static_assert(std::is_same_v<std::decay_t<decltype(custom)>, customize::customize_t>)`

or backward compatible way, check `std::string_view` too.